lib/dataset/bins.cpp

```diff
@@ -140,7 +140,11 @@ DataArray resize_default_init(const DataArray &parent, const Dim dim,
 
 Dataset resize_default_init(const Dataset &parent, const Dim dim,
                             const scipp::index size) {
-  Dataset buffer;
+  auto new_sizes = parent.sizes();
+  if (new_sizes.contains(dim))
+    new_sizes.resize(dim, size);
+
+  Dataset buffer({}, Coords(new_sizes, {}));
```

Comment:
This change predates `setDataInit`. And I don't think it would really benefit from that function. We would still need to insert the coords if the input dataset had no data. And that requires setting the sizes ahead of time. (Unless we also add `setCoordInit`.)